### PR TITLE
Docs: Mention behavior change of splitByChar() in v22.11

### PR DESCRIPTION
**Pull request by rschu1ze**
See #42673

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 2a8e2a0882bf46d3c6db682d78db2b027b3f2517 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54484/2a8e2a0882bf46d3c6db682d78db2b027b3f2517/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/en/sql-reference/functions/splitting-merging-functions.md [hunk 1]
```diff
@@ -33,6 +33,13 @@ Returns an array of selected substrings. Empty substrings may be selected when:
 
 Type: [Array](../../sql-reference/data-types/array.md)([String](../../sql-reference/data-types/string.md)).
 
+:::note
+The behavior of parameter `max_substrings` changed starting with ClickHouse v22.11. In versions older than that, `max_substrings` > 0 meant that `max_substring`-many splits were performed and that the remainder of the string was returned as the final element of the list.
+For example,
+- in v22.10: `SELECT splitByChar('=', 'a=b=c=d', 2); -- ['a','b','c=d']`
```
Comment (nickitat) [inline docs/en/sql-reference/functions/splitting-merging-functions.md:39]: why did we change it? the old behaviour is pretty common across other implementations.

Comment (rschu1ze) [reply docs/en/sql-reference/functions/splitting-merging-functions.md:39]: It was changed in PR #42673, see the discussion after https://github.com/ClickHouse/ClickHouse/pull/42673#issuecomment-1701190114

**Review by nickitat — APPROVED** [review]
🤷🏼

**Comment by aadant** [comment]
@nickitat @rschu1ze : changing an existing behavior breaks users program sometimes silently and makes upgrading CH less straightforward.  You should at the very least introduce a setting or a parameter that preserves the old behavior.

**Comment by nickitat** [comment]
Maybe we could add a compatibility setting

**Comment by rschu1ze** [comment]
@aadant @jhenhapl @UnamedRus @nickitat  --> PR #54518

**Pull request merged**